zerver/lib/users.py

```diff
@@ -501,11 +501,6 @@ def get_api_key(user_profile: UserProfile) -> str:
     return user_profile.api_key
 
 
-def get_all_api_keys(user_profile: UserProfile) -> list[str]:
```

Comment:
This helper was, I'm pretty sure, preparation for #17939, which I think some folks have been asking for. I would probably not remove it unless it's in the way of other work, since we will need to add it when we do that.

Reply:
IDK, I feel like skipping this commit will merge conflict with the later changes, and it's not a lot of code to add again later when we work on that. So I'm fine with just removing it for now.

Reply:
When we implement #17939, the API keys will be a joined table or a method on the model - not as random helper methods in server/lib/users.py.  We will already need to sweep for places that assume a singleton API keys, so this doesn't make any real amount of more work then.